esp/tools/soapplus/httpproxy.cpp

```diff
@@ -660,12 +659,11 @@ class CSocksProxyThread : public Thread
             char peername[256];
             int clientport = m_client->peer_name(peername, 256);
 
-            char inbuf[1024];
-            char outbuf[1024];
-            memset(inbuf, 0, 1024);
-            memset(outbuf, 0, 1024);
+            char inbuf[8];
+            char outbuf[8];
+            memset(inbuf, 0, 8);
+            memset(outbuf, 0, 8);
 
-            unsigned int len = 0;
             unsigned int lenread = 0;
             m_client->read(inbuf, 8, 8, lenread);
```

Comment:
Because min is 8 I don't think this routine will ever return less than 8.  
It may throw with JSOCKERR_timeout_expired or some other socket error.  
If it does throw, the run() will return 0, which is different than this -1.

Reply:
That's true, the if statement is kind pointless here.